### PR TITLE
Add QueueTable component to explain#show

### PR DESCRIPTION
**Pull request by Jeremy1026**
### Description
Adds a QueueTable component to the Explain#Show route for better data display.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
Uncomment `binding.pry` in the `spec/fixes/investigate_duplicate_judge_assign_task_spec.rb` or `explain_spec.rb` RSpec and click on the "Narrative table" link in the browser tabs with the `explain` page. 
Test the filters for the Category and Type columns. Under the 'Details' column, click on any of the numbers to expand the attributes (the number indicates the number of attributes).

### CODE REVIEW

File: client/app/explain/index.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  contextColumn,
+  objectTyleColumn,
+  eventTypeColumn,
+  objectIdColumn,
+  commentColumn,
+  relevanttDataColumn,
+  detailsColumn
+} from './components/ColumnBuilder'
+
+class Explain extends React.PureComponent {
```
Comment (yoomlam) [inline client/app/explain/index.jsx:16]: Can this be more descriptive, like `ExplainEventsTable`?

File: client/app/explain/index.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      ),
+      [EXPLAIN_CONFIG.COLUMNS.EVENT_TYPE.name]: eventTypeColumn(
+        column
+      ),
+      [EXPLAIN_CONFIG.COLUMNS.OBJECT_ID.name]: objectIdColumn(
+        column
+      ),
+      [EXPLAIN_CONFIG.COLUMNS.COMMENT.name]: commentColumn(
+        column
+      ),
```
Comment (yoomlam) [inline client/app/explain/index.jsx:36]: Let's rename the "comment" column to be "narrative". I'm open to an alternative column header.

**Review by yoomlam — COMMENTED** [review]
Some things I noticed.

File: lib/helpers/intake_renderer.rb [hunk 3]
```diff
@@ -252,7 +252,7 @@ def user_details(user)
   end
 
   def truncate(text, size)
-    (text.size > size) ? text[0, size - 1] + "…" : text
+    (text.size > size) ? text[0, size - 1] + "…" : text if text
```
Comment (yoomlam) [inline lib/helpers/intake_renderer.rb:255]: Handles nil text

File: lib/helpers/sanitized_json_configuration.rb [hunk 4]
```diff
@@ -240,7 +240,7 @@ def self.appeals_associated_with(appeal)
     [
       appeal.cavc_remand&.source_appeal,
       appeal.appellant_substitution&.source_appeal,
-      appeal.request_issues.map { |rqi| rqi.contested_decision_issue&.decision_review }
+      appeal.request_issues.includes(:contested_decision_issue).map { |rqi| rqi.contested_decision_issue&.decision_review }
```
Comment (yoomlam) [inline lib/helpers/sanitized_json_configuration.rb:243]: eager load as recommended by `bullet`

File: client/app/explain/components/NarrativeTable.jsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  detailsColumn
+} from './ColumnBuilder';
+import Modal from '../../components/Modal';
+import COPY from '../../../COPY';
+import { css } from 'glamor';
+import PropTypes from 'prop-types';
+
+class NarrativeTable extends React.PureComponent {
+  constructor(props) {
+    super(props);
```
Comment (yoomlam) [inline client/app/explain/components/NarrativeTable.jsx:19]: My IDE says `super` is deprecated. Should this component be converted into a functional component?

File: client/app/explain/components/ColumnBuilder.jsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        {rowData[column.name]} ({rowData.object_id})
+      </span>;
+    },
+    backendCanSort: true,
+    label: 'Sort by comment',
+    getSortValue: (rowData) => rowData[column.name]
+  };
+};
+
+export const relevanttDataColumn = (column) => {
```
Comment (yoomlam) [inline client/app/explain/components/ColumnBuilder.jsx:109]:
```suggestion
export const relevantDataColumn = (column) => {
```

Comment (Jeremy1026) [reply client/app/explain/components/ColumnBuilder.jsx]: Fixed spelling

File: client/app/explain/components/NarrativeTable.jsx (hunk 5 above)
Comment (Jeremy1026) [reply client/app/explain/components/NarrativeTable.jsx:19]: This is the pattern we use across 116 files at the moment, and I don't see us using anything else for building the state. I think we should follow the existing pattern, and maybe have a future PI task to update this throughout the application.

File: client/app/explain/components/ColumnBuilder.jsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+      </span>;
+    },
+    backendCanSort: true,
+    filterOptions,
+    tableData: narratives,
+    columnName: 'category',
+    enableFilter: true,
+    anyFiltersAreSet: true,
+    label: 'Sort by category',
+    bodyClassName: 'closestRegionalOffice.location_hash.city',
```
Comment (yoomlam) [inline client/app/explain/components/ColumnBuilder.jsx:61]: Delete?
```suggestion
```

**Comment by yoomlam** [comment]
image

**Review by yoomlam — APPROVED** [review]

**Pull request merged**